client/download.go

```diff
@@ -138,6 +138,21 @@ func downloadFile(httpClient httpClient, filepath string, url string, devnull bo
 		}
 	}
 
+	size, err = downloadFile(httpClient, temppath, url, devnull, expectedSha)
+	if err != nil {
+		return size, err
+	}
+
+	if !devnull {
```

Comment:
If `devnull` is `true` then we should delete the file IMHO.

Reply:
good point...
`devnull` really complicate this code :wink:

Reply:
Is it even needed to have `devnull` functionality? Wouldn't be better to remove it?